### PR TITLE
Remove FrameChange::document_ready

### LINKED ISSUE #14693: FrameChange::document_ready is never set
`FrameChange::document_ready` is initialized `false` and never set `true`. We can remove it.

Code: `components/constellation/constellation.rs`

### PR DESCRIPTION
**Pull request by zaynetro**
Remove unused `FrameChange::document_ready` field and remove unused boolean in `ConstellationMsg::GetPipeline`'s second argument.

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #14693

- [ ] There are tests for these changes OR
- [x] These changes do not require tests because compiler validates the field is not used anywhere

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/14724)

### CODE REVIEW

File: components/constellation/constellation.rs [hunk 1]
```diff
@@ -1929,15 +1920,15 @@ impl<Message, LTF, STF> Constellation<Message, LTF, STF>
...
 
     fn handle_get_pipeline(&mut self, frame_id: Option<FrameId>,
-                           resp_chan: IpcSender<Option<(PipelineId, bool)>>) {
+                           resp_chan: IpcSender<Option<PipelineId>>) {
         let frame_id = frame_id.unwrap_or(self.root_frame_id);
         let current_pipeline_id = self.frames.get(&frame_id)
             .map(|frame| frame.current.pipeline_id);
         let current_pipeline_id_loaded = current_pipeline_id
-            .map(|id| (id, true));
+            .map(|id| id);
```
Comment (fflorent) [inline components/constellation/constellation.rs]: Sorry for the late question: what this `map` call does? As I read it, the callback is an identity function, so it just copies the array.

NB: I am a beginner in Rust, so I may be wrong, don't hesitate to correct me :).

Comment (zaynetro) [inline components/constellation/constellation.rs]: Good catch, it looks like it is not needed anymore. (I may be wrong as well :) )

Comment (Ms2ger) [inline components/constellation/constellation.rs]: A PR to remove `current_pipeline_id_loaded` would be welcome :)

Comment (asajeffrey) [inline components/constellation/constellation.rs]: Yes, this used to do something, back when it was map(|id| (id, true)). Now it's pointless. Good catch.

**Pull request merged**